samples/src/main/kotlin/com/github/fsbarata/functional/samples/FormValidation.kt

```diff
@@ -0,0 +1,59 @@
+package com.github.fsbarata.functional.samples
+
+import com.github.fsbarata.functional.data.list.NonEmptyList
+import com.github.fsbarata.functional.data.list.concatNelSemigroup
+import com.github.fsbarata.functional.data.validation.Validation
+import com.github.fsbarata.functional.data.validation.sequence
+import java.util.regex.Pattern
+
+typealias Email = String
+typealias PhoneNumber = String
+
+sealed class ValidationError {
+    object InvalidMail : ValidationError()
+    object InvalidPhoneNumber : ValidationError()
+}
+
+private fun Email.toValidatedEmail(): Validation<NonEmptyList<ValidationError>, Email> =
+    when {
+        isValidEmail() -> Validation.Success(this)
+        else -> Validation.Failure(ValidationError.InvalidMail).toValidationNel()
```

Comment:
Can also be Validation.Failure(nelOf(InvalidMail))

Reply:
Done